httpclient-backend/monix/src/main/scala/sttp/client/httpclient/monix/MonixWebSocketHandler.scala

```diff
@@ -1,149 +1,36 @@
 package sttp.client.httpclient.monix
 
-import java.net.http.WebSocket.Listener
-import java.net.http.{WebSocket => JWebSocket}
-import java.nio.ByteBuffer
-import java.util.concurrent.atomic.AtomicBoolean
-import java.util.concurrent.{CompletableFuture, CompletionStage}
-import java.util.function.BiConsumer
-
+import monix.catnap.MVar
 import monix.eval.Task
 import monix.execution.Scheduler
 import sttp.client.httpclient.WebSocketHandler
-import sttp.client.impl.monix.{MonixAsyncQueue, TaskMonadAsyncError}
-import sttp.client.monad.syntax._
-import sttp.client.monad.{Canceler, MonadAsyncError, MonadError}
+import sttp.client.httpclient.internal.NativeWebSocketHandler
+import sttp.client.impl.monix.TaskMonadAsyncError
 import sttp.client.ws.internal.AsyncQueue
 import sttp.client.ws.{WebSocket, WebSocketEvent}
-import sttp.model.ws.{WebSocketClosed, WebSocketFrame}
 
 object MonixWebSocketHandler {
 
   /**
     * Returns an effect, which creates a new [[WebSocketHandler]]. The handler should be used *once* to send and
     * receive from a single websocket.
-    *
-    * @param incomingBufferCapacity Amount of messages which will be buffered on client side before backpressure kicks in
-    * Default value is 73 because 73 is the 21th prime number, its mirror number is the 12th prime number,
-    * who’s mirror number 21 is the product of 7*3. Furthermore, 73 is 1001001 in binary, which is a palindrome.
     */
-  def apply(incomingBufferCapacity: Int = 73)(implicit
+  def apply()(implicit
       s: Scheduler
   ): Task[WebSocketHandler[WebSocket[Task]]] = {
-    require(
-      incomingBufferCapacity >= 2,
-      "Incoming buffer capacity has to be at least 2 (opening frame + one data frame)"
-    )
     Task {
-      val isOpen: AtomicBoolean = new AtomicBoolean(false)
-      val queue = new MonixAsyncQueue[WebSocketEvent](Some(incomingBufferCapacity))
-      WebSocketHandler(
-        new AddToQueueListener(queue, isOpen, incomingBufferCapacity),
-        httpClientWebSocketToWebSocket(_, queue, isOpen, TaskMonadAsyncError)
-      )
+      val queue = new SingleElementQueue[WebSocketEvent]
```

Comment:
thinking about this, we probably do need a 2-element queue (that is, we can still use `NativeWebSocketHandler`, just with a different queue).

That's because of error and close events. If a data frame is waiting to be received, and the ws is closed or there's a network error, despite there being no frames requested (as the data frame is not yet received, and we haven't called `ws.request(1)`), a `Close`/`Error` event will be enqueued. So in this case, we'll have two elements.

And when using an `MVar`, `.put` will block the network thread, which shouldn't happen.

Reply:
Good point... I wonder if we could write a test for it

Reply:
No idea :) we would have to test that the network thread isn't blocked? 

Maybe ... a server which sends a "welcome message" when the WS is opened - this should block, then sending should maybe block as well? - that is, the task would never complete.

Maybe we could try with the `/ws/send_and_wait` test endpoint - it sends two messages when the ws is open